### PR TITLE
Update docker build and fix ARM64 image

### PR DESCRIPTION
**Pull request by hakman**
Closes: https://github.com/kubernetes/node-problem-detector/issues/797

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -71,15 +71,15 @@ else ifeq ($(shell go env GOHOSTOS), windows)
 ENABLE_JOURNALD=0
 endif
 
-# TODO(random-liu): Support different architectures.
-# The debian-base:v2.0.0 image built from kubernetes repository is based on
-# Debian Stretch. It includes systemd 241 with support for both +XZ and +LZ4
-# compression. +LZ4 is needed on some os distros such as COS.
-BASEIMAGE:=registry.k8s.io/build-image/debian-base:bullseye-v1.4.3
+# Set default base image to Debian 12.1 (Bookworm)
+BASEIMAGE:=debian:bookworm-20230814-slim@sha256:741bae561f5c2261f4cdd535e4fd4c248dec0aafc1b9a1410b3d67ad24571340
```
Comment (pacoxu) [inline Makefile:75]: kubernetes/etcd is using `registry.k8s.io/build-image/debian-base:bookworm-v1.0.0` in https://github.com/kubernetes/kubernetes/blob/04292dd94bd9999dfced146f4de8b81a0e466e06/cluster/images/etcd/Makefile#L95

Comment (hakman) [reply Makefile:75]: Is there any advantage to using `registry.k8s.io/build-image/debian-base`?

Comment (pacoxu) [reply Makefile:75]: As I know, we always avoid docker hub as folks will run into rate limits.

Comment (hakman) [reply Makefile:75]: This is just for runtime image, to build things. This is pushed to registry.k8s.io later.

Comment (hakman) [reply Makefile:75]: Also, the etcd image is more or less (to my knowledge) for  internal testing. Not sure if there's any released image based on `registry.k8s.io/build-image/debian-base`.

Comment (tzneal) [reply Makefile:75]: Any reason not to use the same base image?  A few more advantages to using it are that future updates will likely occur at the same time (e.g. someone updating registry.k8s.io/build-image/debian-base:bookworm-v1.0.0 will find all of the occurrences) and less network/disk usage by downstream users due to the shared base image.

Comment (hakman) [reply Makefile:75]: In my opinion, a build image should be used for that and not for final artifacts. In this case, the image contains the Go toolkit and all its dependencies, plus other scripts, that should not be shipped.
For other kube images, there are different base images, see https://github.com/kubernetes/kubernetes/tree/master/build/server-image.
NPD is not tied to a specific K8s release so, the base image will probably be different between various components. 

In the end, `debian:bookworm-slim` is 30MB in size. Performance is not an issue at this size.
Updating the base image should also be easier by not depending on upstream builds and releases.

Comment (hakman) [reply Makefile:75]: @tzneal @pacoxu Any thoughts on the above?

Comment (tzneal) [reply Makefile:75]: My opinion is that moving from a registry.k8s.io hosted base image to a non-registry.k8s.io hosted base image requires a really good reason.

Comment (vteratipally) [reply Makefile:75]: > 

+1 to this.

Comment (hakman) [reply Makefile:75]: Would `registry.k8s.io/build-image/go-runner` be ok instead?
https://github.com/search?q=org%3Akubernetes+registry.k8s.io%2Fbuild-image%2Fgo-runner&type=code

Comment (hakman) [reply Makefile]: I tried using `debian-base:bookworm-v1.0.0`, but for some reason tests failed with it and had to revert to `debian-base:bullseye-v1.4.3`. Tests are passing consistently now, but the base image contains know CVEs, as mentioned in https://github.com/kubernetes/node-problem-detector/issues/800.

**Comment by hakman** [comment]
@tzneal @vteratipally Could you take one more look? Thanks!

**Comment by hakman** [comment]
Tests are now passing consistently, especially since https://github.com/kubernetes/test-infra/pull/30710 was merged.

**Comment by vteratipally** [comment]
/lgtm
/approve

Thank you for the fix.

**Comment by pacoxu** [comment]
@vteratipally k/k tried to upgrade to the last release of NPD and reverted due to this issue. Will NPD release a new version in recent days? Or what is the release plan of NPD?

**Comment by vteratipally** [comment]
We can release a new version this week to fix the issue

**Comment by hakman** [comment]
@vteratipally @pacoxu maybe we want https://github.com/kubernetes/node-problem-detector/pull/814 too in the release, to be on the same Go version as k/k.

**Pull request merged**